templates/home/index.html.twig

```diff
@@ -1,7 +1,61 @@
 {% extends 'base.html.twig' %}
 
-{% block title %}{{ someVariable }}{% endblock %}
+{% block title %}Projektai{% endblock %}
 
 {% block body %}
-    <div class="nfq-academy">{{ someVariable }}</div>
+
+    <ul class="list-group m-4">
+        <li class="list-group-item list-group-item-info">Studentai</li>
+        {% for project in data %}
+            {% for student in project.students %}
+                    <li class="list-group-item">
+                        <a href="{{ path('student') }}?name={{ student }}&amp;project={{ project.name|replace({' ': ''})|lower }}"> {{ student }}</a>
+                        {% for mentor in project.mentors %}
+                            (<span class="badge">Mentorius</span> {{ mentor }})
+                        {% endfor %}
+                    </li>
+            {% endfor %}
+        {% endfor %}
+    </ul>
+
+    <div class="m-4">
+        <form action="{{ path('student') }}" method="get">
+            <div class="input-group">
+                <input name="name" type="text" class="form-control" placeholder="Studentas"/>
+            </div>
+            <div class="input-group">
+                <input name="project" type="text" class="form-control" placeholder="Projektas"/>
+            </div>
+            <button type="submit" class="btn btn-success">Sužinoti vertinimą</button>
+        </form>
+    </div>
+
+    <ul class="list-group m-4">
+        <li class="list-group-item list-group-item-info">Projektai</li>
+        {% for project in data %}
+            {% set projectName = project.name|replace({' ': ''})|lower %}
```

Comment:
Kurti naujus kintamuosius Twig'e (`set` funkcija) nėra gera praktika. Twig'e turėtų būti kiek įmanoma mažiau logikos.
Nes PHP pusę yra legviau pratestuoti (kontroluoti aplinką), negu užsikraudinėti visą Twig dizainą.

Taigi geresnė praktika yra turėti pvz. (privačią) funkciją `groupByStudents` PHP pusėje, kad ir tame pačiame kontroleryje.

Plačiau:
 * https://symfony.com/doc/current/testing.html (akademijoje bus atskira paskaita tam)
 * https://stackoverflow.com/questions/17026405/twig-template-unit-testing

Reply:
Taisydama atsisakiau  kintamojo "projectName" twige, kadangi pritaikiau `{% for key, project in data %}`. Reikiamą reikšmę pasiekiau per `key`.